drivers/shared/executor/utils.go

```diff
@@ -75,7 +75,10 @@ func CreateExecutor(
 	return newExecutorClient(config, logger)
 }
 
-// ReattachToExecutor launches a plugin with a given plugin config
+// ReattachToExecutor launches a plugin with a given plugin config and validates it can call the executor.
+// On Windows, go-plugin listens on a localhost port. It is possible on a reboot that another process is
+// listening on that port, and a process is running with the previous executors PID, leading Nomad to kill
+// "random" processes. Fail early via the Version RPC if we detect the listener isn't actually an Executor.
```

Comment:
Great addition to the comment. Were you able to to track down where the kill was being emitted downstream? It might be nice to mention that here, as I know both of us spent a good bit of time trying to find it.

Reply:
In my reproduction I noticed the downstream call happening [here in the taskRunner](https://github.com/hashicorp/nomad/blob/main/client/allocrunner/taskrunner/task_runner.go#L707) after a failed RPC happened.  I will update the comment to add this context.